### PR TITLE
Add internal dns and consul variable parser

### PR DESCRIPTION
**Pull request by gdenn**
Added a feature so the INTERNAL_DNS_IP and CONSUL_IP can be parsed from the bosh director using `bosh -d consul-dns --json`.

### CODE REVIEW

File: spec/lib/unit_tests_utils/consul_spec.rb [hunk 1]
```diff
@@ -3,16 +3,18 @@
 
 describe UnitTestsUtils::Consul do
   let(:internal_consul_ip) { "1.2.3.4" }
-
-  before :each do
```
Comment (lucaspinto) [inline spec/lib/unit_tests_utils/consul_spec.rb]: You could do it like in InternalDNS where you left this one and just overwrote it in one case.

I do not have a preference between both solutions but I think you should pick just one unless there's a reason I am overseeing.

Comment (gdenn) [inline spec/lib/unit_tests_utils/consul_spec.rb]: If i got you right, you want that we make the before each and after each hooks consistent. I commited brackets into every before and after hook here https://github.com/anynines/unit-tests-utils/pull/3/commits/957a42c46c806bef473039e13e6e985d773e36fe .

File: lib/unit_tests_utils/consul.rb [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+      instances = JSON.parse(`bosh vms -d consul-dns --json`)["Tables"][0]["Rows"]
+
+      instances.each_with_index do |val, index| 
+        if instances[index]["instance"].match?("consul") then 
+          return instances[index]["ips"] 
+        end 
+      end
+    end
+
+    raise 'No consul ip found'
```
Comment (lucaspinto) [inline lib/unit_tests_utils/consul.rb]: "No IP found for Consul."

File: lib/unit_tests_utils/internal_dns.rb [hunk 3]
```diff
@@ -14,7 +16,19 @@ def self.host_addresses(hostnames, port)
...
+      instances = JSON.parse(`bosh vms -d consul-dns --json`)["Tables"][0]["Rows"]
+      
+      instances.each_with_index do |val, index| 
+        if instances[index]["instance"].match?("dnsmasq") then 
+          return instances[index]["ips"] 
+        end 
+      end
+    end
+
+    raise 'No internal dns ip found'
```
Comment (lucaspinto) [inline lib/unit_tests_utils/internal_dns.rb]: "No Internal DNS IP found."

File: spec/lib/unit_tests_utils/consul_spec.rb (hunk 1 above)
Comment (lucaspinto) [inline spec/lib/unit_tests_utils/consul_spec.rb]: I was not refering to the brackets.

**Pull request closed without merging**